hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieIOFactory.java

```diff
@@ -48,4 +48,13 @@ private static HoodieIOFactory getIOFactory(String ioFactoryClass) {
 
   public abstract HoodieFileWriterFactory getWriterFactory(HoodieRecord.HoodieRecordType recordType);
 
+  public abstract HoodieStorage getStorage(StoragePath storagePath);
+
+  public abstract HoodieStorage getStorage(StoragePath path,
```

Comment:
How much effort is required to remove this API?

Reply:
Maybe we can just pass `FileSystemRetryConfig`? I am not very familiar with what this is

Reply:
OK. We take this on separately.